### PR TITLE
Rename ecma_get_number to ecma_op_to_number

### PR DESCRIPTION
**Pull request by orkvi**
JerryScript-DCO-1.0-Signed-off-by: Virag Orkenyi orkvi@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/api/jerry.c [hunk 1]
```diff
@@ -1269,7 +1269,9 @@ jerry_value_to_number (const jerry_value_t value) /**< input value */
     return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
   }
 
-  return jerry_return (ecma_op_to_number (value));
+  ecma_number_t num;
+  ecma_op_to_number (value, &num);
```
Comment (rerobika) [inline jerry-core/api/jerry.c:1273]: Missing error handling.

File: jerry-core/ecma/builtin-objects/ecma-builtin-json.c [hunk 2]
```diff
@@ -1256,13 +1256,15 @@ ecma_builtin_json_serialize_property (ecma_json_stringify_context_t *context_p,
     /* 5.a */
     if (class_name == LIT_MAGIC_STRING_NUMBER_UL)
     {
-      value = ecma_op_to_number (value);
+      ecma_number_t num;
+      ecma_op_to_number (value, &num);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-json.c:1260]: ditto.

File: jerry-core/ecma/builtin-objects/ecma-builtin-number.c [hunk 3]
```diff
@@ -65,7 +65,9 @@ ecma_builtin_number_dispatch_call (const ecma_value_t *arguments_list_p, /**< ar
   }
   else
   {
-    ret_value = ecma_op_to_number (arguments_list_p[0]);
+    ecma_number_t num;
+    ecma_op_to_number (arguments_list_p[0], &num);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-number.c:69]: ditto(error handling).

File: jerry-core/ecma/operations/ecma-array-object.c [hunk 4]
```diff
@@ -950,31 +950,27 @@ ecma_op_array_object_set_length (ecma_object_t *object_p, /**< the array object
...
-    ecma_value_t compared_num_val = ecma_op_to_number (new_value);
+    ecma_value_t compared_num_val = ecma_op_to_number (new_value, &new_len_num);
 
     if (ECMA_IS_VALUE_ERROR (compared_num_val))
     {
       return compared_num_val;
     }
 
-    new_len_num = ecma_get_number_from_value (compared_num_val);
     ecma_free_value (compared_num_val);
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-array-object.c:974]: Freeing this value is unnecessary.

File: jerry-core/ecma/operations/ecma-arraybuffer-object.c [hunk 5]
```diff
@@ -119,15 +119,12 @@ ecma_op_create_arraybuffer_object (const ecma_value_t *arguments_list_p, /**< li
...
-
+      ecma_value_t to_number_value = ecma_op_to_number (arguments_list_p[0], &length_num);
       if (ECMA_IS_VALUE_ERROR (to_number_value))
       {
         return to_number_value;
       }
 
-      length_num = ecma_get_number_from_value (to_number_value);
-
       ecma_free_value (to_number_value);
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-arraybuffer-object.c:128]: Ditto.

File: jerry-core/ecma/operations/ecma-comparison.c [hunk 6]
```diff
@@ -112,13 +112,15 @@ ecma_op_abstract_equality_compare (ecma_value_t x, /**< first operand */
...
 
       if (ECMA_IS_VALUE_ERROR (x_num_value))
       {
         return x_num_value;
       }
-
+      x_num_value = ecma_make_number_value (num);
       ecma_value_t compare_result = ecma_op_abstract_equality_compare (x_num_value, y);
 
       ecma_free_value (x_num_value);
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-comparison.c:126]: Ditto.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 7]
```diff
@@ -330,7 +261,7 @@ ecma_op_to_number (ecma_value_t value) /**< ecma value */
  *         Returned value must be freed with ecma_free_value
  */
 ecma_value_t
-ecma_get_number (ecma_value_t value, /**< ecma value*/
+ecma_op_to_number (ecma_value_t value, /**< ecma value*/
                  ecma_number_t *number_p) /**< [out] ecma number */
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:265]: misaligned line.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 8]
```diff
@@ -215,11 +216,14 @@ ecma_builtin_math_object_hypot (const ecma_value_t *arg, /**< arguments list */
...
+      ecma_number_t num0;
+
+      ecma_value_t value = ecma_op_to_number (*arg, &num0);
       if (ECMA_IS_VALUE_ERROR (value))
       {
         return value;
       }
+      value = ecma_make_number_value (num0);
       arg_num = ecma_get_number_from_value (value);
       ecma_fast_free_value (value);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c:228]: Ditto.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 9]
```diff
@@ -357,13 +361,14 @@ ecma_builtin_math_dispatch_routine (uint16_t builtin_routine_id, /**< built-in w
...
 
         if (ECMA_IS_VALUE_ERROR (value))
         {
           return value;
         }
-
+        value = ecma_make_number_value (num);
         x = ecma_get_number_from_value (value);
 
         ecma_fast_free_value (value);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c:374]: No value release is needed.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 10]
```diff
@@ -357,13 +361,14 @@ ecma_builtin_math_dispatch_routine (uint16_t builtin_routine_id, /**< built-in w
       }
       else
       {
-        ecma_value_t value = ecma_op_to_number (arguments_list[0]);
+        ecma_number_t num;
+        ecma_value_t value = ecma_op_to_number (arguments_list[0], &num);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c:365]: No `num` is needed, you can use `x`.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 11]
```diff
@@ -379,13 +384,14 @@ ecma_builtin_math_dispatch_routine (uint16_t builtin_routine_id, /**< built-in w
       }
       else
       {
-        ecma_value_t value = ecma_op_to_number (arguments_list[1]);
+        ecma_number_t num1;
+        ecma_value_t value = ecma_op_to_number (arguments_list[1], &num1);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c:388]: ditto, just for `y`.

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 12]
```diff
@@ -291,14 +291,15 @@ ecma_builtin_string_object_from_code_point (ecma_value_t this_arg, /**< 'this' a
 
   for (ecma_length_t index = 0; index < args_number; index++)
   {
-    ecma_value_t to_number_value = ecma_op_to_number (args[index]);
+    ecma_number_t num;
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c:294]: no `num` is needed, you can use `to_number_num` below.

File: jerry-core/vm/vm.c [hunk 13]
```diff
@@ -2659,12 +2659,14 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           }
           else
           {
-            result = ecma_op_to_number (left_value);
+            ecma_number_t num;
```
Comment (rerobika) [inline jerry-core/vm/vm.c:2662]: `num` -> `result_number` below.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 14]
```diff
@@ -135,13 +135,14 @@ ecma_builtin_math_object_max_min (bool is_max, /**< 'max' or 'min' operation */
...
-      ecma_value_t value = ecma_op_to_number (*arg);
+      ecma_number_t num;
+      ecma_value_t value = ecma_op_to_number (*arg, &num);
 
       if (ECMA_IS_VALUE_ERROR (value))
       {
         return value;
       }
-
+      value = ecma_make_number_value (num);
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c:145]: This is not necessary, just pass `&arg_num` to ecma_op_to_number.

File: jerry-core/ecma/builtin-objects/ecma-builtin-math.c [hunk 15]
```diff
@@ -215,13 +216,16 @@ ecma_builtin_math_object_hypot (const ecma_value_t *arg, /**< arguments list */
...
+      ecma_number_t num0;
+
+      ecma_value_t value = ecma_op_to_number (*arg, &num0);
       if (ECMA_IS_VALUE_ERROR (value))
       {
         return value;
       }
-      arg_num = ecma_get_number_from_value (value);
-      ecma_fast_free_value (value);
+      ecma_value_t num_value = ecma_make_number_value (num0);
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-math.c:226]: Same here.

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 16]
```diff
@@ -291,19 +291,20 @@ ecma_builtin_string_object_from_code_point (ecma_value_t this_arg, /**< 'this' a
 
   for (ecma_length_t index = 0; index < args_number; index++)
   {
-    ecma_value_t to_number_value = ecma_op_to_number (args[index]);
+    ecma_number_t to_number_num;
+    ecma_value_t to_number_value = ecma_op_to_number (args[index], &to_number_num);
```
Comment (dbatyai) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c:295]: This could be removed, the `ecma_op_to_integer` call below will call to_number anyways.

File: jerry-core/ecma/operations/ecma-array-object.c [hunk 17]
```diff
@@ -950,32 +950,26 @@ ecma_op_array_object_set_length (ecma_object_t *object_p, /**< the array object
...
-
-  ecma_number_t new_len_num = ecma_get_number_from_value (completion);
+  JERRY_ASSERT (!ECMA_IS_VALUE_ERROR (completion));
 
   ecma_free_value (completion);
 
   if (ecma_is_value_object (new_value))
   {
-    ecma_value_t compared_num_val = ecma_op_to_number (new_value);
+    ecma_value_t compared_num_val = ecma_op_to_number (new_value, &new_len_num);
```
Comment (dbatyai) [inline jerry-core/ecma/operations/ecma-array-object.c:967]: This entire `if (ecma_is_value_object (new_value))` block is not necessary, the previous ecma_op_to_number will already handle objects.

File: jerry-core/ecma/base/ecma-helpers-number.c [hunk 18]
```diff
@@ -765,7 +765,7 @@ ecma_number_parse_int (const lit_utf8_byte_t *string_buff, /**< routine's first
 
   /* 6. */
   ecma_number_t radix_num;
-  radix = ecma_get_number (radix, &radix_num);
+  radix = ecma_op_to_number (radix, &radix_num);
 
   if (!ecma_is_value_empty (radix))
```
Comment (rerobika) [inline jerry-core/ecma/base/ecma-helpers-number.c:770]:
```c
if (ECMA_IS_VALUE_ERROR (ecma_op_to_number (radix, &radix_num)))
{
  return ECMA_VALUER_ERROR;
}
```

File: jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.c [hunk 19]
```diff
@@ -407,14 +407,12 @@ ecma_builtin_date_prototype_dispatch_set (uint16_t builtin_routine_id, /**< buil
...
-    ecma_value_t value = ecma_op_to_number (arguments_list[i]);
+    ecma_value_t value = ecma_op_to_number (arguments_list[i], &converted_number[i]);
 
     if (ECMA_IS_VALUE_ERROR (value))
     {
       return value;
     }
-
-    converted_number[i] = ecma_get_number_from_value (value);
     ecma_free_value (value);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.c:416]: This value should not be freed.

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.c [hunk 20]
```diff
@@ -785,7 +786,7 @@ ecma_builtin_date_dispatch_construct (const ecma_value_t *arguments_list_p, /**<
           return prim_value;
         }
 
-        prim_value_num = ecma_date_time_clip (ecma_get_number_from_value (prim_value));
+        prim_value_num = ecma_date_time_clip (arg);
 
         ecma_free_value (prim_value);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.c:791]: ditto.

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c (hunk 16 above)
Comment (rerobika) [reply jerry-core/ecma/builtin-objects/ecma-builtin-string.c:295]: This request is still valid.

File: jerry-core/ecma/operations/ecma-array-object.c [hunk 21]
```diff
@@ -950,32 +950,26 @@ ecma_op_array_object_set_length (ecma_object_t *object_p, /**< the array object
...
     return completion;
   }
 
-  JERRY_ASSERT (!ECMA_IS_VALUE_ERROR (completion)
-                && ecma_is_value_number (completion));
-
-  ecma_number_t new_len_num = ecma_get_number_from_value (completion);
+  JERRY_ASSERT (!ECMA_IS_VALUE_ERROR (completion));
 
   ecma_free_value (completion);
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-array-object.c:963]: This should not be freed.

File: jerry-core/ecma/operations/ecma-array-object.c (hunk 17 above)
Comment (rerobika) [reply jerry-core/ecma/operations/ecma-array-object.c:967]: This request is still valid.

File: jerry-core/vm/vm.c (hunk 13 above)
Comment (rerobika) [reply jerry-core/vm/vm.c:2662]: This request is still valid.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 22]
```diff
@@ -935,7 +867,35 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
...
 
+  if (ecma_number_is_nan (value) || ecma_number_is_infinity (value))
+  {
+    return ECMA_VALUE_FALSE;
+  }
+
+  ecma_number_t floor_fabs = floor (fabs (value));
+  ecma_number_t fabs_value = fabs (value);
+
+  if (floor_fabs != fabs_value)
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:892]: `return ecma_make_boolean_value (floor_fabs == fabs_value)`

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/base/ecma-helpers-number.c [hunk 23]
```diff
@@ -765,7 +765,12 @@ ecma_number_parse_int (const lit_utf8_byte_t *string_buff, /**< routine's first
 
   /* 6. */
   ecma_number_t radix_num;
-  radix = ecma_op_to_numeric (radix, &radix_num, ECMA_TO_NUMERIC_NO_OPTS);
+  radix = ecma_op_to_number (radix, &radix_num);
```
Comment (dbatyai) [inline jerry-core/ecma/base/ecma-helpers-number.c:768]: The result of this is checked twice, on line 770 and on line 775. We should apply @rerobika's suggestion:
> ```c
> if (ECMA_IS_VALUE_ERROR (ecma_op_to_number (radix, &radix_num)))
> {
>   return ECMA_VALUER_ERROR;
> }
> ```

File: jerry-core/ecma/operations/ecma-array-object.c (hunk 17 above)
Comment (dbatyai) [reply jerry-core/ecma/operations/ecma-array-object.c:939]: Bump.

**Review by dbatyai — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/base/ecma-helpers-number.c [hunk 24]
```diff
@@ -765,13 +765,13 @@ ecma_number_parse_int (const lit_utf8_byte_t *string_buff, /**< routine's first
...
 
-  if (!ecma_is_value_empty (radix))
+  if (ECMA_IS_VALUE_ERROR (radix))
   {
-    return radix;
+    return ECMA_VALUE_ERROR;
   }
-
+/**/
```
Comment (dbatyai) [inline jerry-core/ecma/base/ecma-helpers-number.c:774]: Please remove this comment.

**Review by dbatyai — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 25]
```diff
@@ -943,7 +875,30 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
 
   return ret_value;
 } /* ecma_op_to_property_descriptor */
+/**
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:878]: New line above.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 26]
```diff
@@ -943,7 +875,30 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
...
+ * IsInteger operation.
+ *
+ * See also:
+ *          ECMA-262 v5, 9.4
+ *          ECMA-262 v6, 7.1.4
+ *
+ * @return true if the argument is
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:885]:
```
@return true - if the argument is integer
              false - otherwise
```

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 27]
```diff
@@ -943,7 +875,30 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
...
+ *         integer false otherwise
+ */
+ecma_value_t
+ecma_is_integer (ecma_number_t value) /**< ecma value */
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:889]: `ecma_is_integer` -> `ecma_op_is_integer`

`(ecma_number_t value) /**< ecma value */` -> `(ecma_number_t num) /**< ecma number */`

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 28]
```diff
@@ -943,7 +875,30 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
...
+ * IsInteger operation.
+ *
+ * See also:
+ *          ECMA-262 v5, 9.4
+ *          ECMA-262 v6, 7.1.4
+ *
+ * @return true if the argument is
+ *         integer false otherwise
+ */
+ecma_value_t
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:888]: This should return a simple `bool`

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 29]
```diff
@@ -943,7 +875,30 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
...
+ecma_is_integer (ecma_number_t value) /**< ecma value */
+{
+
+  if (ecma_number_is_nan (value) || ecma_number_is_infinity (value))
+  {
+    return ECMA_VALUE_FALSE;
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:894]: `return false`

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 30]
```diff
@@ -943,7 +875,30 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
...
+  }
 
+  ecma_number_t floor_fabs = floor (fabs (value));
+  ecma_number_t fabs_value = fabs (value);
+
+  return ecma_make_boolean_value (floor_fabs == fabs_value);
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:900]: ecma_make_boolean_value is not needed.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 31]
```diff
@@ -943,7 +875,30 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
...
+} /* ecma_is_integer*/
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:901]: Comment should be updated.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 32]
```diff
@@ -975,20 +930,19 @@ ecma_op_to_integer (ecma_value_t value, /**< ecma value */
   ecma_number_t number = *number_p;
 
   /* 3 */
+
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:933]: Please revert this change.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 33]
```diff
@@ -975,20 +930,19 @@ ecma_op_to_integer (ecma_value_t value, /**< ecma value */
...
   if (ecma_number_is_nan (number))
   {
     *number_p = ECMA_NUMBER_ZERO;
     return ECMA_VALUE_EMPTY;
   }
-
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:983]: Ditto.

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 34]
```diff
@@ -975,20 +930,19 @@ ecma_op_to_integer (ecma_value_t value, /**< ecma value */
...
   /* 4 */
   if (ecma_number_is_zero (number) || ecma_number_is_infinity (number))
   {
     return ECMA_VALUE_EMPTY;
   }
 
   ecma_number_t floor_fabs = floor (fabs (number));
-
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:991]: Ditto.

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/operations/ecma-conversion.c [hunk 35]
```diff
@@ -278,77 +278,9 @@ ecma_op_to_boolean (ecma_value_t value) /**< ecma value */
  */
 ecma_value_t
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-conversion.c:279]: One more thing:

This should be an `inline` function with `JERRY_ATTR_ALWAYS_INLINE` attribute.

**Review by rerobika — CHANGES_REQUESTED** [review]

**Review by rerobika — APPROVED** [review]
LGTM

**Review by dbatyai — APPROVED** [review]
LGTM
